yamux/src/connection.rs

```diff
@@ -352,6 +356,8 @@ struct Active<T> {
     stream_receiver: mpsc::Receiver<StreamCommand>,
     dropped_streams: Vec<StreamId>,
     pending_frames: VecDeque<Frame<()>>,
+    pending_acks: HashSet<StreamId>,
```

Comment:
When a stream fails for some reason, is it removed from `pending_acks`?

Also, in case the remote closes a stream right away, i.e. does not write to it, does the remote set the ACK bit and is the stream thus removed from `pending_acks`?

Reply:
Hmm, good questions.

> Also, in case the remote closes a stream right away, i.e. does not write to it, does the remote set the ACK bit and is the stream thus removed from `pending_acks`?

You mean sending `FIN`? I don't think that is allowed by the spec. A new stream needs to either be acknowledged or reset.

I don't I am removing them on reset though, let me add that.